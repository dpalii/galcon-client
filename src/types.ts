export enum OutgoingEvents {
    CREATE_NEW_GAME = 'CREATE_NEW_GAME',
    JOIN_GAME = 'JOIN_GAME',
    START_GAME = 'START_GAME',
    SEND_UNITS = 'SEND_UNITS',
}

export enum IncomingEvents {
    PLAYER_JOINED = 'PLAYER_JOINED',
    GAME_STARTED = 'GAME_STARTED',
    SEND_UNITS = 'GAME_STARTED',
    WON = 'WON',
    LOST = 'LOST',
}

export interface GameDetails {
    gameId: string
    player1: string
    player2: string
}

export interface PlanetData {
    coords: Coordinates
    fleet: number
    fleetGenSpeed: number
    id: number
    ownerId: string
    radius: number
}

export interface MapData {
    w: number
    h: number
    planetArray: PlanetData[]
}

export interface GameState {
    id: string
    map: MapData
    player1: string
    player2: string
}

export interface Coordinates {
    x: number
    y: number
}
