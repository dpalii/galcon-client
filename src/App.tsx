import './App.css';
import { ConnectModal } from './components/ConnectModal/ConnectModal';
import { Game } from './components/Game/Game';
import { useContext, useState } from 'react';
import { SocketContext } from './contexts/SocketContext';
import { GameDetails, IncomingEvents, OutgoingEvents } from './types';

function App() {
  const socket = useContext(SocketContext);
  const [gameId, setGameId] = useState<string | null>(null);
  // const [players, setPlayers] = useState<string[]>([]);
  const handleConnect = (gameId: string) => {
    socket.emit(OutgoingEvents.JOIN_GAME, gameId, (response: GameDetails) => {
      console.log(response.gameId);
      setGameId(response.gameId);
      // setPlayers([response.player1, response.player2]);
    });
  }
  const handleNewGame = () => {
    socket.emit(OutgoingEvents.CREATE_NEW_GAME, (response: string) => {
      console.log(response);
      setGameId(response);
      // setPlayers([]);
    });
  }
  socket.on(IncomingEvents.PLAYER_JOINED, (...args) => {
    console.log(args);
  });

  return (
    <div className="App">
      {
        gameId
          ? <Game gameId={gameId} /> 
          : <ConnectModal onConnect={handleConnect} onCreateNewGame={handleNewGame} />
      }
    </div>
  );
}

export default App;
