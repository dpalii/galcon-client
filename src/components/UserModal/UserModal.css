.error {
    color: #b00
}

#name {
    flex: 1;
    border: none;
    background-color: var(--bg-secondary);
    padding: 10px 15px;
    border-radius: 4px;
}

#name:focus {
    outline: none;
}

.modal label {
    width: 50px;
    padding-right: 10px;
    text-align: left;
}

.input-wrapper {
    display: flex;
    flex-direction: row;
    padding-bottom: 10px;
}
