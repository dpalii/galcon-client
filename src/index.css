:root {
    font-size: 16px;
    --primary-color: #000;
    --secondary-color: #777;
    --bg-primary: #fff;
    --bg-secondary: #ddd;
    --text-success: #eee;
    --bg-success: #090;
    --bg-success-dark: #070;
    --bg-success-darker: #050;
    --text-default: #fff;
    --bg-default: #33d;
    --bg-default-dark: #22c;
    --bg-default-darker: #11b;
    --text-error: #fff;
    --bg-disabled: #999;
}

body {
    margin: 0;
    font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', 'Roboto', 'Oxygen',
        'Ubuntu', 'Cantarell', 'Fira Sans', 'Droid Sans', 'Helvetica Neue',
        sans-serif;
    -webkit-font-smoothing: antialiased;
    -moz-osx-font-smoothing: grayscale;
}

code {
    font-family: source-code-pro, Menlo, Monaco, Consolas, 'Courier New',
        monospace;
}