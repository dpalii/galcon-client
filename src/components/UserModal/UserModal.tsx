import { useState } from 'react'
import { InputUser } from '../../types'
import './UserModal.css'

export interface UserModalProps {
    inputUser: InputUser
    createGame: (user: InputUser) => void
    joinByCode: (user: InputUser) => void
    openLobbyList: (user: InputUser) => void
}

export function UserModal({ inputUser, createGame, joinByCode, openLobbyList }: UserModalProps) {
    const [error, setError] = useState('')
    const [user, setUser] = useState<InputUser>(inputUser);

    const isFormValid = () => {
        if (!user.name || !user.color) {
            setError('Please fill out the fields')
            return false
        }
        setError('')
        return true
    }

    const handleCreateGame = () => {
        if (!isFormValid()) {
            return
        }
        createGame(user)
    }

    const handleJoinByCode = () => {
        if (!isFormValid()) {
            return
        }
        joinByCode(user)

    }

    const handleOpenLobbyList = () => {
        if (!isFormValid()) {
            return
        }
        openLobbyList(user)
    }

    return (
        <div className="backdrop">
            <div className="modal" onClick={(e) => e.stopPropagation()}>
                <h2 className="heading">USER CONFIGURATION</h2>
                <form>
                    <div className='input-wrapper'>
                        <label htmlFor="name">Name</label>
                        <input
                            id="name"
                            name="name"
                            value={user.name}
                            onChange={(e) =>
                                setUser({ ...user, name: e.target.value })
                            }
                        />
                    </div>
                    <div className='input-wrapper'>
                        <label htmlFor="color">Color</label>
                        <input
                            id="color"
                            name="color"
                            type="color"
                            value={user.color}
                            onChange={(e) =>
                                setUser({ ...user, color: e.target.value })
                            }
                        />
                    </div>
                </form>
                <p className='error'>{error}</p>
                <button
                    className="btn mb w-100"
                    onClick={() => handleCreateGame()}
                >
                    CREATE GAME
                </button>
                <button
                    className="btn mb w-100"
                    onClick={() => handleJoinByCode()}
                >
                    JOIN BY CODE
                </button>
                <button
                    className="btn mb w-100"
                    onClick={() => handleOpenLobbyList()}
                >
                    OPEN LOBBY LIST
                </button>
            </div>
        </div>
    )
}