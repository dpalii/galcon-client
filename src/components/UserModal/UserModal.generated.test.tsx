/* eslint-disable testing-library/no-node-access */
/* eslint-disable testing-library/no-unnecessary-act */
import { render } from "react-dom";
import { fireEvent} from '@testing-library/react'

import { InputUser } from '../../../src/types'

import { UserModal } from '../../../src/components/UserModal/UserModal';

import { unmountComponentAtNode } from "react-dom";
import { act } from 'react-dom/test-utils';

let container: any = null;

beforeEach(() => {
  // setup a DOM element as a render target
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  // cleanup on exiting
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

describe('UserModal', () => {
  it('should render component', () => {
    act(() => {
      render(<UserModal  
        inputUser={{name: '', color: ''}} 
        createGame={(user: InputUser) => {}} 
        joinByCode={ (user: InputUser) => {}} 
        openLobbyList={(user: InputUser) => {}} />, container);
    });
    
    expect(container).toMatchSnapshot();
  })

  it('empty user', () => {
    act(() => {
      render(<UserModal  
        inputUser={{name: '', color: ''}} 
        createGame={(user: InputUser) => {}} 
        joinByCode={ (user: InputUser) => {}} 
        openLobbyList={(user: InputUser) => {}} />, container);
    });

    container.querySelector('#create-button').dispatchEvent(new MouseEvent("click", { bubbles: true }));
    container.querySelector('#join-button').dispatchEvent(new MouseEvent("click", { bubbles: true }));
    container.querySelector('#open-button').dispatchEvent(new MouseEvent("click", { bubbles: true }));
    
    expect(container).toMatchSnapshot();
  })

  it('events', () => {
    act(() => {
      render(<UserModal  
        inputUser={{name: '', color: ''}} 
        createGame={(user: InputUser) => {}} 
        joinByCode={ (user: InputUser) => {}} 
        openLobbyList={(user: InputUser) => {}} />, container);
    });

    container.querySelector(["[id='create-button']"]).dispatchEvent(new MouseEvent("click", { bubbles: true }));
    fireEvent.change(container.querySelector('#name'), { target: { value: 'name' } });
    fireEvent.change(container.querySelector('#color'), { target: { value: '#000000' } });
    container.querySelector('#create-button').dispatchEvent(new MouseEvent("click", { bubbles: true }));
    container.querySelector('#join-button').dispatchEvent(new MouseEvent("click", { bubbles: true }));
    container.querySelector('#open-button').dispatchEvent(new MouseEvent("click", { bubbles: true }));
    expect(container).toMatchSnapshot();
  })
});