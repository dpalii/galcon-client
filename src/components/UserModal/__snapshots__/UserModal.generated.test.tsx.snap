// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserModal empty user 1`] = `
<div>
  <div
    class="backdrop"
  >
    <div
      class="modal"
    >
      <h2
        class="heading"
      >
        USER CONFIGURATION
      </h2>
      <form>
        <div
          class="input-wrapper"
        >
          <label
            for="name"
          >
            Name
          </label>
          <input
            id="name"
            name="name"
            value=""
          />
        </div>
        <div
          class="input-wrapper"
        >
          <label
            for="color"
          >
            Color
          </label>
          <input
            id="color"
            name="color"
            type="color"
            value=""
          />
        </div>
      </form>
      <p
        class="error"
      >
        Please fill out the fields
      </p>
      <button
        class="btn mb w-100"
        id="create-button"
      >
        CREATE GAME
      </button>
      <button
        class="btn mb w-100"
        id="join-button"
      >
        JOIN BY CODE
      </button>
      <button
        class="btn w-100"
        id="open-button"
      >
        OPEN LOBBY LIST
      </button>
    </div>
  </div>
</div>
`;

exports[`UserModal events 1`] = `
<div>
  <div
    class="backdrop"
  >
    <div
      class="modal"
    >
      <h2
        class="heading"
      >
        USER CONFIGURATION
      </h2>
      <form>
        <div
          class="input-wrapper"
        >
          <label
            for="name"
          >
            Name
          </label>
          <input
            id="name"
            name="name"
            value="name"
          />
        </div>
        <div
          class="input-wrapper"
        >
          <label
            for="color"
          >
            Color
          </label>
          <input
            id="color"
            name="color"
            type="color"
            value="#000000"
          />
        </div>
      </form>
      <p
        class="error"
      />
      <button
        class="btn mb w-100"
        id="create-button"
      >
        CREATE GAME
      </button>
      <button
        class="btn mb w-100"
        id="join-button"
      >
        JOIN BY CODE
      </button>
      <button
        class="btn w-100"
        id="open-button"
      >
        OPEN LOBBY LIST
      </button>
    </div>
  </div>
</div>
`;

exports[`UserModal should render component 1`] = `
<div>
  <div
    class="backdrop"
  >
    <div
      class="modal"
    >
      <h2
        class="heading"
      >
        USER CONFIGURATION
      </h2>
      <form>
        <div
          class="input-wrapper"
        >
          <label
            for="name"
          >
            Name
          </label>
          <input
            id="name"
            name="name"
            value=""
          />
        </div>
        <div
          class="input-wrapper"
        >
          <label
            for="color"
          >
            Color
          </label>
          <input
            id="color"
            name="color"
            type="color"
            value=""
          />
        </div>
      </form>
      <p
        class="error"
      />
      <button
        class="btn mb w-100"
        id="create-button"
      >
        CREATE GAME
      </button>
      <button
        class="btn mb w-100"
        id="join-button"
      >
        JOIN BY CODE
      </button>
      <button
        class="btn w-100"
        id="open-button"
      >
        OPEN LOBBY LIST
      </button>
    </div>
  </div>
</div>
`;
